demo

```diff
@@ -211,45 +233,72 @@ done
+            ${DOCKER_COMPOSE_PROJECT_NAME}_mender-useradm_1 \
+            /usr/bin/useradm create-user \
+            --username=${USER} \
+            --password=${PASSWORD} \
+            > /dev/null
     retval=$?
-    case $retval in
-        0)
-            echo "****************************************"
-            echo
-            echo "Username: mender-demo@example.com"
-            echo  "Login password: ${PASSWORD}"
-            echo
-            echo "****************************************"
-            echo "Please keep the password available, it will not be cached by the login script."
-            break ;;
-        5) # If the user exists, skip uploading the Artifact
-            UPLOAD_ARTIFACT=""
-            echo "The user already exists. Skipping"
-            echo "If you don't remember the password, you can run '$(basename $0) down' to delete"
-            echo "the old user and rerun '$(basename $0) up' to create a new one."
-            echo "Please note that all data will be deleted from the old demo server."
-            break ;;
-        *) echo "docker exec error: " $retval; exit $retval ;;
-    esac
-    if [[ $RETRIES -ge $RETRY_LIMIT ]]; then
-        echo "Retried $RETRIES times without success. Giving up."
-        exit 1
+    EXISTS_ERROR=5
+fi
+if [ $retval -eq 0 ]; then
+    PRINT_LOGIN_INFO=1
+elif [ $retval -eq $EXISTS_ERROR ]; then
+    # If the user exists, skip uploading the Artifact
+    UPLOAD_ARTIFACT=0
+    PRINT_USER_EXISTS=1
+else
+    echo "docker exec error: " $retval
+    exit $retval
+fi
+
+if [ $ENTERPRISE = 1 -a $CLIENT = 1 ]; then
+    if [ $(docker ps -q -f name=${DOCKER_COMPOSE_PROJECT_NAME}_mender-client_1 | wc -l) -gt 0 ]; then
+        # If already launched, we don't need to do anything.
+        :
+    elif [ -n "$TENANT_ID" ]; then
+        TENANT_TOKEN=$(docker exec \
+                ${DOCKER_COMPOSE_PROJECT_NAME}_mender-tenantadm_1 \
+                /usr/bin/tenantadm get-tenant \
+                --id $TENANT_ID \
+                | jq -r .tenant_token)
+        # Now that we have the tenant token we can enable the client.
+        EXTRA_FILES="$EXTRA_FILES $CLIENT_FILES"
+        TENANT_TOKEN=$TENANT_TOKEN docker-compose \
+                -f docker-compose.yml \
+                -f docker-compose.storage.minio.yml \
+                -f docker-compose.demo.yml \
+                -p ${DOCKER_COMPOSE_PROJECT_NAME} \
+                $EXTRA_FILES \
+                "$@" -d mender-client
+    else
+        echo "WARNING: Ignoring request to launch the Mender client."
+        echo "In Enterprise mode, the client can only be launched the first time the server"
+        echo "is started, when the first user is created. If you wish to start from scratch,"
+        echo "replace \`up\` with \`down -v\` first to reset, then rerun."
     fi
-    RETRIES=$((RETRIES+1))
-    sleep 5
-done
+fi
 
-if [[ $UPLOAD_ARTIFACT == "true" ]]; then
+if [[ $UPLOAD_ARTIFACT -eq 1 ]]; then
```

Comment:
For consistency: use `[ ` here too.

Reply:
I'm trying to fix this, but I'm a bit unsure about this one:

```bash
if [[ "$@" == *"-f docker-compose.no-ssl.yml"* ]]
```

I understand what it's trying to do, but this looks like incorrect syntax to me. I can't find a reference in the bash documentation that wild cards actually work on arbitrary strings (except in `case` that is). AFAIK they only work on file names that are on disk. And in that context, is `[[` important, or not? @lluiscampos, @oleorhagen: Do any of you know?

Reply:
Hmm, yes, this one struck me as a bit odd also. How about first:
` "$@" -> "$*"` ?
Then:
```bash
if [[ "$*" =~ <regex> ]]; then
```
The intended use-case above I think needs `extglob` enabled: https://www.gnu.org/software/bash/manual/html_node/The-Shopt-Builtin.html to enable this sort of pattern matching.

However, I might be wrong...

Reply:
Hmm, the documentation for that option is quite confusing, it doesn't really explain whether the parentheses are part of the expression or just a way to denote the syntax. And there are zero examples... I think I will just rewrite it into a `case` match instead, which is a bit more well known, and doesn't require a special `shopt` option.

Reply:
Sounds good to me :)

Reply:
Decided to leave it as is after discussing with @oleorhagen.

Reply:
Regarding, use of `[[` and `[` in general, @oleorhagen convinced me. The double variant is less portable, but let's stick with it as long as we know we are in bash, since it is more well behaved with regards to quoting, and has more matching options.